frontend/src/scenes/session-recordings/playlist/sessionRecordingsPlaylistLogic.ts

```diff
@@ -1,128 +1,680 @@
+            (activeSessionRecording, recordings, autoplayDirection): Partial<SessionRecordingType> | undefined => {
+                if (!activeSessionRecording || !autoplayDirection) {
+                    return
+                }
+                const activeSessionRecordingIndex = recordings.findIndex((x) => x.id === activeSessionRecording.id)
+                return autoplayDirection === 'older'
+                    ? recordings[activeSessionRecordingIndex + 1]
+                    : recordings[activeSessionRecordingIndex - 1]
+            },
+        ],
+        hasNext: [
+            (s) => [s.sessionRecordingsResponse],
+            (sessionRecordingsResponse) => sessionRecordingsResponse.has_next,
+        ],
+        totalFiltersCount: [
+            (s) => [s.filters, (_, props) => props.personUUID],
+            (filters, personUUID) => {
+                const defaultFilters = getDefaultFilters(personUUID)
+
+                return (
+                    (filters?.actions?.length || 0) +
+                    (filters?.events?.length || 0) +
+                    (filters?.properties?.length || 0) +
+                    (equal(filters.session_recording_duration, defaultFilters.session_recording_duration) ? 0 : 1) +
+                    (filters.date_from === defaultFilters.date_from && filters.date_to === defaultFilters.date_to
+                        ? 0
+                        : 1) +
+                    (filters.console_logs?.length || 0)
+                )
+            },
+        ],
+        hasAdvancedFilters: [
+            (s) => [s.filters, (_, props) => props.personUUID],
+            (filters, personUUID) => {
+                const defaultFilters = getDefaultFilters(personUUID)
+                return addedAdvancedFilters(filters, defaultFilters)
+            },
+        ],
+
+        otherRecordings: [
+            (s) => [s.sessionRecordings, s.hideViewedRecordings, s.pinnedRecordings, s.selectedRecordingId],
+            (
+                sessionRecordings,
+                hideViewedRecordings,
+                pinnedRecordings,
+                selectedRecordingId
+            ): SessionRecordingType[] => {
+                return sessionRecordings.filter((rec) => {
+                    if (pinnedRecordings.find((pinned) => pinned.id === rec.id)) {
+                        return false
+                    }
+
+                    if (hideViewedRecordings && rec.viewed && rec.id !== selectedRecordingId) {
+                        return false
+                    }
+
+                    return true
+                })
+            },
+        ],
+
+        recordings: [
+            (s) => [s.pinnedRecordings, s.otherRecordings],
+            (pinnedRecordings, otherRecordings): SessionRecordingType[] => {
+                return [...pinnedRecordings, ...otherRecordings]
+            },
+        ],
+    }),
+
+    actionToUrl(({ props, values }) => {
```

Comment:
Actually you can probably ignore all comments on linking to a playlist - I assume the behaviour moved to here

Reply:
I might actually move this into the "scene" logics as each scene is slightly different and the "updateSearchParams" just makes things more complicated I would argue...